src/rdiff_backup/rpath.py

```diff
@@ -766,7 +766,11 @@ def setmtime(self, modtime):
         except OSError:
             # It's not possible to set a modification time for
             # directories on Windows.
-            if self.conn.os.name != 'nt' or not self.isdir():
+            if Globals.get_api_version() < 201:  # compat200
```

Comment:
As another tough. Should we not have a fs_abilities to check if say directory support modification date.

So we have a test at the beginning only and use the result for the rest of the backup. Current implementation will make the same call to `self.conn.platform.system()` for every directory. Right ?

Reply:
The idea is good. Actually, I should at least check first if it's a directory, and then only check which OS it is, that would already reduce the number of calls across the line.
Your suggestion requires more changes, let me think about it.

Reply:
So, I've implemented the more efficient approach, checking first for a directory and _then_ only for Windows.

Additionally, I've created two enhancement requests #795 and #796 - but it's something for 2.4 not for now.